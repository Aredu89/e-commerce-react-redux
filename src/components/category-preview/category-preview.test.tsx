import { screen } from '@testing-library/react';
import { renderWithStoreAndRouter } from '../../utils/test/test.utils';

import CategoryPreview from './category-preview.component';

describe('Category preview', () => {
  let props = {
    title: 'Title',
    items: [
      {
        id: 1,
        imageUrl: 'image/url',
        name: 'Item name',
        price: 100,
      }
    ]
  };
  it('renders properly', () => {
    const result = renderWithStoreAndRouter(<CategoryPreview {...props} />);
    expect(result).toMatchSnapshot();
  });
  it('shows the title', () => {
    renderWithStoreAndRouter(<CategoryPreview {...props} />);
    expect(screen.getByText('TITLE')).toBeInTheDocument();
  });
  it('renders only 4 items', () => {
    props = {
      ...props,
      items: [
        {
          id: 1,
          imageUrl: 'image/url',
          name: 'Item name',
          price: 100,
        },
        {
          id: 2,
          imageUrl: 'image/url',
          name: 'Item name 2',
          price: 100,
        },
        {
          id: 3,
          imageUrl: 'image/url',
          name: 'Item name 3',
          price: 100,
        },
        {
          id: 4,
          imageUrl: 'image/url',
          name: 'Item name 4',
          price: 100,
        },
        {
          id: 5,
          imageUrl: 'image/url',
          name: 'Item name 5',
          price: 100,
        },
      ]
    };
    renderWithStoreAndRouter(<CategoryPreview {...props} />);
    expect(screen.getAllByText('ADD TO CART').length).toBe(4);
  });
});
