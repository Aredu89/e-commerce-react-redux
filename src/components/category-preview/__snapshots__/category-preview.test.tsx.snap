// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Category preview renders properly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-bczRLJ feYkiB"
      >
        <a
          class="sc-gsnTZi cVjYOm"
          href="/Title"
        >
          TITLE
        </a>
        <div
          class="sc-dkzDqf dXpcTy"
        >
          <div
            class="sc-jqUVSM bJZKRg"
          >
            <img
              alt="Item name"
              src="image/url"
            />
            <div
              class="sc-papXJ eDnHsY"
            >
              <span
                class="sc-kDDrLX gaInqx"
              >
                Item name
              </span>
              <span
                class="sc-iqcoie edOgOJ"
              >
                $
                100
              </span>
            </div>
            <button
              class="sc-jSMfEi sc-iBkjds ctEdUo fLgvBh"
            >
              ADD TO CART
            </button>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-bczRLJ feYkiB"
    >
      <a
        class="sc-gsnTZi cVjYOm"
        href="/Title"
      >
        TITLE
      </a>
      <div
        class="sc-dkzDqf dXpcTy"
      >
        <div
          class="sc-jqUVSM bJZKRg"
        >
          <img
            alt="Item name"
            src="image/url"
          />
          <div
            class="sc-papXJ eDnHsY"
          >
            <span
              class="sc-kDDrLX gaInqx"
            >
              Item name
            </span>
            <span
              class="sc-iqcoie edOgOJ"
            >
              $
              100
            </span>
          </div>
          <button
            class="sc-jSMfEi sc-iBkjds ctEdUo fLgvBh"
          >
            ADD TO CART
          </button>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
